### PR TITLE
Added callable options for iteration_log and epoch_log in StatsHandler

### LINKED ISSUE #5964: overriding iteration_log and epoch_log in StatsHandler
Currently, they are booleans but we could extend them to `Union[bool, Callable]` where if arg is callable then it could be used for event filtering:

```python
    def attach(self, engine):


         if self.iteration_log and not engine.has_event_handler(self.iteration_completed, Events.ITERATION_COMPLETED):
             event_filter = self.iteration_log if callable(self.iteration_log) else None
             engine.add_event_handler(
                 Events.ITERATION_COMPLETED(event_filter=event_filter), self.iteration_completed
             ) 
         if self.epoch_log and not engine.has_event_handler(self.epoch_completed, Events.EPOCH_COMPLETED):
             event_filter = self.epoch_log if callable(self.epoch_log) else None
             engine.add_event_handler(
                 Events.EPOCH_COMPLETED(event_filter=event_filter), self.epoch_completed
             ) 
```

_Originally posted by @vfdev-5 in https://github.com/Project-MONAI/MONAI/discussions/5958#discussioncomment-4912997_

### PR DESCRIPTION
**Pull request by vfdev-5**
Fixes #5964

### Description

Added callable options for iteration_log and epoch_log in StatsHandler.

Ref: https://github.com/Project-MONAI/MONAI/discussions/5958#discussioncomment-4912997

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by wyli — APPROVED** [review]

File: tests/test_handler_stats.py [hunk 1]
```diff
@@ -26,74 +26,97 @@
...
-        log_handler = logging.StreamHandler(log_stream)
-        log_handler.setLevel(logging.INFO)
-        key_to_handler = "test_logging"
-        key_to_print = "testing_metric"
+        def event_filter(_, event):
+            if event in [1, 2]:
+                return True
+            return False
+
+        for epoch_log in [True, event_filter]:
```
Comment (vfdev-5) [inline tests/test_handler_stats.py:34]: @wyli can we use something like pytest parametrize here instead of for-loop ?

Comment (wyli) [reply tests/test_handler_stats.py:34]: thanks, I can think of this approach: https://github.com/vfdev-5/MONAI/pull/1

**Comment by vfdev-5** [comment]
@wyli I wonder which job actually runs test_handler_stats.py, can you please put a link on it ?

**Comment by wyli** [comment]
I think this one https://github.com/Project-MONAI/MONAI/actions/runs/4134447826/jobs/7145596787

File: monai/handlers/stats_handler.py [hunk 2]
```diff
@@ -135,9 +141,19 @@ def attach(self, engine: Engine) -> None:
...
+                else Events.ITERATION_COMPLETED
+            )
+            engine.add_event_handler(event, self.iteration_completed)
         if self.epoch_log and not engine.has_event_handler(self.epoch_completed, Events.EPOCH_COMPLETED):
-            engine.add_event_handler(Events.EPOCH_COMPLETED, self.epoch_completed)
+            event = (
+                Events.EPOCH_COMPLETED(event_filter=self.epoch_log)
+                if callable(self.epoch_log)
+                else Events.EPOCH_COMPLETED
+            )
```
Comment (ericspod) [inline monai/handlers/stats_handler.py:155]:
```suggestion
            event = Events.EPOCH_COMPLETED
            if callable(self.epoch_log)  # substitute event with new one using filter callable
                event = event(event_filter=self.epoch_log)  
```

**Comment by ericspod** [comment]
I had a few minor suggestions but looks like a good change!

**Review by Nic-Ma — APPROVED** [review]
Hi @vfdev-5 ,

Looks good to me.
Could you please help also enable this feature for other very similar handlers? Like:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/handlers/tensorboard_handlers.py#L94
And:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/handlers/mlflow_handler.py#L101

Thanks in advance.

**Comment by vfdev-5** [comment]
@Nic-Ma yes, I can take a look on that in a follow-up PR

**Pull request merged**